Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1245,6 +1245,11 @@ public IImmutableSet<TxId> GetStagedTransactionIds()
 
                     Store.AppendIndex(Id, block.Hash);
 
+                    foreach (var tx in block.Transactions)
```

Comment:
```suggestion
                    foreach (Transaction<T> tx in block.Transactions)
```

Reply:
Isn't it straightforward for you to guess the Type of the variable `tx` for this statement?

When you want to cast the right value into the left value, you'd better declare the Type explicitly and let the compiler do implicit casting. Or If it is very hard for a code reviewer to guess the Type, you'd better do that.

However
IMHO,  in many cases, for static type language, It is recommended to use `var` instead of declaring Type explicitly. 
If there is something that I missed, please teach me. And please put that rule into the coding rule.